internal/workload/manager.go

```diff
@@ -10,47 +10,22 @@ import (
 
 	"git.sr.ht/~spc/go-log"
 	api2 "github.com/jakub-dzon/k4e-device-worker/internal/workload/api"
-	"github.com/jakub-dzon/k4e-device-worker/internal/workload/network"
-	podman2 "github.com/jakub-dzon/k4e-device-worker/internal/workload/podman"
 	"github.com/jakub-dzon/k4e-operator/models"
 	v1 "k8s.io/api/core/v1"
 	"sigs.k8s.io/yaml"
 )
 
-const nfTableName string = "edge"
-
 type WorkloadManager struct {
 	manifestsDir string
 	workloads    *workloadWrapper
 }
 
-// workloadWrapper manages the workload and its configuration on the device
-type workloadWrapper struct {
-	workloads *podman2.Podman
-	netfilter *network.Netfilter
-}
-
-func newWorkloadWrapper() (*workloadWrapper, error) {
-	newPodman, err := podman2.NewPodman()
-	if err != nil {
-		return nil, err
-	}
-	netfilter, err := network.NewNetfilter()
-	if err != nil {
-		return nil, err
-	}
-	return &workloadWrapper{
-		workloads: newPodman,
-		netfilter: netfilter,
-	}, nil
-}
-
 func NewWorkloadManager(configDir string) (*WorkloadManager, error) {
 	manifestsDir := path.Join(configDir, "manifests")
 	if err := os.MkdirAll(manifestsDir, 0755); err != nil {
 		return nil, fmt.Errorf("cannot create directory: %w", err)
 	}
-	wrapper, err := newWorkloadWrapper()
+	wrapper, err := newWorkloadWrapper(configDir)
```

Comment:
Do you think we need to execute this one only [once](https://golang.org/src/sync/once.go) to avoid any attempt to open the files more than once during the execution of the worker ?

Reply:
No, I don't think it's needed - the initialization of the components can't happen more than once during device-workers instance life.